TaskWithInheritanceAndPolymorphism/TaskWithInheritanceAndPolymorphism.cpp

```diff
@@ -0,0 +1,139 @@
+﻿#include <iostream>
+#include <memory>
+#include <vector>
+
+class Figure
+{
+public:
+	Figure()
+	{
+		std::cout << "Figure()" << std::endl;
+	}
+	~Figure()
+	{
+		std::cout << "~Figure()" << std::endl;
+	}
+
+	virtual const void what() = 0;
+	virtual const double area() = 0;
+	virtual const double perimeter() = 0;
+};
+
+class Square final : public Figure
+{
+public:
+	Square(const double l)
+	{
+		m_length = l;
+		std::cout << "Square()" << std::endl;
+	}
+	virtual ~Square()
+	{
+		std::cout << "~Square()" << std::endl;
+	}
+
+	void const what() override
+	{
+		std::cout << "I'm Square" << std::endl;
+	}
+	double const area() override
+	{
+		double area = m_length * m_length;
+		std::cout << area << std::endl;
+		return area;
+	}
+	double const perimeter() override
+	{
+		double perimeter = 4 * m_length;
+		std::cout << perimeter << std::endl;
+		return perimeter;
+	}
+private:
+	double m_length;
```

Comment:
Please declare the field like const double m_length. Use a member initializer list in the constructor. Your constructor will be: `Square(const double l) : l(length) { ...}`

Reply:
Fixed